index.html

```diff
@@ -173,126 +173,447 @@ <h2>
+                                <a href="https://dom.spec.whatwg.org/#concept-node-document">node document</a>.
+                              </li>
+                              <li>
+                                The <a>promise</a> must be resolved.
+                              </li>
+                              <li>
+                                Return <a>promise</a>.
+                              </li>
+                            </ol>
+                        </li>
+                      </ol>
+                    </ol>
+                  </li>
+                  <li>
+                    If <a>options</a> are not equivalent to the current <a>pointer-lock options</a> and 
+                    <a>options</a> are supported then perform the following steps
+                    <ol>
+                      <li>
+                        Update the user agent's <a>pointer-lock target</a> to this element.
+                      </li>
+                      <li>
+                        Update the user agent's <a>pointer-lock options</a> to <a>options</a>
+                      </li>
+                      <li>
+                        <a href="https://html.spec.whatwg.org/multipage/webappapis.html#queue-an-element-task">
+                          Queue an element task</a> on the 
+                          <a href="https://html.spec.whatwg.org/multipage/webappapis.html#user-interaction-task-source">
+                          User Interaction task source</a>, given this, to perform the following steps:
+                          <ol type='i'>
+                            <li>
+                              <a href=https://dom.spec.whatwg.org/#concept-event-fire">Fire an event</a> named
+                              <a>pointerlockerror</a> at <a>this</a>'s 
+                              <a href="https://dom.spec.whatwg.org/#concept-node-document">node document</a>.
+                            </li>
+                            <li>
+                              The <a>promise</a> must be resolved.
+                            </li>
+                            <li>
+                              Return <a>promise</a>.
+                            </li>
+                          </ol>
+                      </li>
+                    </ol>
+                  </li>
+                </ol>
+              </li>
+              <li>
+                If the user agent's current pointer-lock target's shadow-including root is the
+                same as this element's shadow-including root, then:
+                <ol type='i'>
+                  <li>
+                    The user agent's <a>pointer-lock target</a> must be updated to this element
+                  </li>
+                  <li>
+                    a <a>pointerlockchange</a> event must be sent
+                  </li>
+                  <li>
+                    the promise must <a>resolve</a>
+                  </li>
```

Comment:
You probably want to return promise from this branch, instead of continuing onward.